pkg/network/vlan/vlan.go

```diff
@@ -172,18 +175,57 @@ func (v *Vlan) stopMonitor() {
 	network.GetWatcher().DelLink(v.nic.Index())
 }
 
-func (v *Vlan) AddLocalArea(id int) error {
+func (v *Vlan) AddLocalArea(id int, cidr string) error {
 	if v.nic == nil {
 		return fmt.Errorf("physical nic vlan network")
 	}
-	return v.nic.AddBridgeVlan(uint16(id))
+	if id == PVID {
+		return nil
+	}
+
+	if err := v.nic.AddBridgeVlan(uint16(id)); err != nil {
+		return fmt.Errorf("add bridge vlan %d failed, error: %w", id, err)
+	}
+
+	if cidr == "" {
+		return nil
+	}
+
+	if err := iface.EnsureRouteViaGateway(cidr); err != nil {
+		return fmt.Errorf("ensure %s to route via gateway failed, error: %w", cidr, err)
+	}
```

Comment:
We delete routes on vlan setup. Here we add routes again. Does it help if we don't clean up routes at the begining?

Reply:
No, they are different routes.